app/jobs/calendar_importer/parsers/dice_fm.rb

```diff
@@ -5,53 +5,25 @@
 # Parent parser classes should not be added.
 
 module CalendarImporter::Parsers
-  class DiceFm < Base
-    NAME = 'Dice FM'
+  class DiceFm < LdJson
+    NAME = 'Dice FM (LD+JSON)'
     KEY = 'dice-fm'
     DOMAINS = %w[dice.fm].freeze
 
-    def self.whitelist_pattern
+    def self.allowlist_pattern
       %r{^https://dice\.fm/venue/*}
     end
 
-    def download_calendar
-      response = HTTParty.get(@url)
-      return [] unless response.success?
-
-      doc = Nokogiri::HTML(response.body)
-      data_nodes = doc.xpath('//script[@type="application/ld+json"]')
-      return [] if data_nodes.empty?
-
-      [].tap do |raw_event_data|
-        data_nodes.each do |node|
-          data = safely_parse_json(node.inner_html, {})
-
-          case data
-          when Hash
-            raw_event_data << data
-
-          when Array
-            raw_event_data.concat data
-
-          else
-            Rails.logger.debug { "Unrecognised RDF type '#{data.class.name}'" }
-          end
-        end
-      end
+    def initialize(calendar, options = {})
+      options[:consumer_helper] = EventConsumer
+      super calendar, options
     end
 
-    def import_events_from(data)
-      records = JSON::LD::API
-                .expand(data)
-
-      place_record = records.find { |record| record['@type'].first == 'http://schema.org/Place' }
-      return [] if place_record.blank?
-
-      events = place_record['http://schema.org/event']
-
-      events
-        .map { |event_hash| CalendarImporter::Events::DiceFmEvent.new(event_hash) }
-        .keep_if(&:event_record?)
+    module EventConsumer
+      def consume_place(data)
+        Rails.logger.debug 'consume_place'
+        consume data['event']
```

Comment:
What's this doing that isn't done by the `LdJson` importer and why can't the `LdJson` importer do it instead? In an ideal world, we wouldn't have a special "Dice" importer at all, because the `LdJson` one does everything we need.

Reply:
i'll try and explain in more detail with examples when i have a fresher mind but if you're curious OutSavvy tucks their events under a `@graph` node and Dice hides their events in a `Place` record.

Reply:
This is now resolved into one unified ld-json parser